### PR TITLE
Strafing support like Switch CS+

### PR DESCRIPTION
**Pull request by chaoskagami**
A bit of a feature copy - Cave Story+ on the Nintendo Switch has a "feature" where by holding either ZL or ZR Quote will not change direction while moving, allowing one to backstep while firing the weapon at an enemy or lock vertical trajectory to one side with the fireball.

This adds a new key (KEYSTRAFE) and makes it so Quote's direction/look is not changed when it is held. It also adds a setting to make this added feature toggleable, since strafing makes some parts far easier.

Unfortunately, this breaks the settings file since a new key was added as well as saved option for the toggle. The version was incremented (NSX3 -> NSX4) to avoid problems.

Comments requested. The changes may be a bit messy - let me know if I should fix/reorder anything. I think it's a neat feature of the Switch version that's worth adding here, but that's just my personal feelings on the matter.

### CODE REVIEW

File: src/pause/options.cpp [hunk 1]
```diff
@@ -183,6 +185,10 @@ Dialog *dlg = opt.dlg;
 	dlg->AddItem("Tracks: ", _tracks_change, _tracks_get, -1, OD_CHOICE);
 	dlg->AddItem("Sound: ", _sound_change, _sound_get, -1, OD_CHOICE);
 
+	dlg->AddSeparator();
+
+	dlg->AddItem("Cave Story+ Strafing: ", _strafe_change, _strafe_get, -1, OD_CHOICE);
```
Comment (isage) [inline src/pause/options.cpp]:
```suggestion
	dlg->AddItem("Strafing: ", _strafe_change, _strafe_get, -1, OD_CHOICE);
```

**Comment by isage** [comment]
Looks good overall.
This would require additions to translations, but i'll deal with it later, before next release.

File: src/pause/options.cpp (hunk 1 above)
Comment (chaoskagami) [reply src/pause/options.cpp]: Can do. Honestly, I debated whether the input subsection was more appropriate, but it overflows out of the box from there.

**Comment by chaoskagami** [comment]
Force pushed to rename "Cave Story+ Strafing" -> "Strafing:". I also missed a spacing issue, so I corrected that while I was at it.

As for translations...yeah, sorry for making more work for you. I don't think I can help much with that one.  :(

**Pull request merged**